fastify.js

```diff
@@ -135,7 +135,20 @@ function fastify (options) {
   }
 
   // Instance Fastify components
-  const { logger, hasLogger } = createLogger(options)
+
+  // create a default logger and then overwrite if valid logger options provided
+  let logger
+  let hasLogger
+
+  if (options.loggerInstance || options.logger) {
+    const { createLogger } = require('./lib/logger');
+    ({ logger, hasLogger } = createLogger(options))
+  } else {
+    const nullLogger = require('abstract-logging')
+    logger = nullLogger
+    logger.child = () => logger
+    hasLogger = false
+  }
```

Comment:
I thought a bit more and while the solution LGTM, but let me push this code forward:
we are moving some logger logic outside the `/logger` and `/logger-factory` in the `fastify.js` file and, while it works, it does not feel perfect.

Here we could write the old:

```
const { logger, hasLogger } = createLogger(options)
```

Then, in `lib/logger-factory.js` we may expose a `createLogger` function that will look like (dirty code):

```
function createLogger() {
  if (options.loggerInstance || options.logger) {
    const pinoLoggerFactory = require('./lib/logger-pino');
    return pinoLoggerFactory(options)
  } else {
    const nullLogger = require('abstract-logging')
    logger = nullLogger
    logger.child = () => logger
    return { logger, hasLogger: false }
  }
}
```

Where `logger-pino.js` is the old `lib/logger.js`.

This would clean up a bit the code because atm:

- the logger optimization lives in fastify.js
- the pino logger lives in `lib/logger.js`
- the abstract logger lives in fastify.js

With a proposal solution:

- fasitfy.js does not care about logging optimizations
- pino logger lives in `lib/logger-pino.js`
- abstract logger lives in `lib/logger-factory.js` (we may move it to its own file, but for 3 lines of code this would be just silly 😄 )

Who does it sound to you?

Reply:
@Eomm This sounds perfect to me. Let me take it in a separate PR. Also, can you help re-trigger the failed check in this PR as there is some other test which has failed now

Reply:
Thank you for calling all of that out @Eomm. I was also put off by the re-org.

Reply:
Also, can someone please retrigger the failed only test. It seems flaky too.

Reply:
> I will open a new PR with changes suggested by Eomm as it will require a good refactor of code. Does that sound okay?

The changes should be in here, because there is no reason to split the logic then refactor later.

Reply:
@climba03003 @jsumners @mcollina  Have made the changes as per @Eomm suggestions. Please let me know what you guys think.